src/test/java/com/wojciechbarwinski/demo/epic_board_games_shop/services/OrderServiceHelperTest.java

```diff
@@ -0,0 +1,202 @@
+        assertEquals(2, preparedOrder.getOrderLines().size());
+        assertEquals(expectedTotalPrice, preparedOrder.getTotalPrice());
+        assertEquals(OrderStatus.PLACED, preparedOrder.getOrderStatus());
+        assertEquals(employeeId, preparedOrder.getEmployeeId());
+        assertEquals(2, preparedOrder.getOrderLines().get(0).getQuantity());
+    }
+    
+    // whe order quantity is more than in stock
+
+    @Test
+    void shouldThrowExceptionBecauseThereIsNoAuthUserWhenOrderIsPlace() {
+        //given
+        setUpSecurityContextForTests(false);
+        OrderRequestDTO orderDTO = createCorrectOrderRequestDTO();
+        Order order = createOrderInStageAfterMapping();
+        String expectedMessage = "No authenticated user found.";
+
+        when(productRepository.findAllById(Mockito.any())).thenReturn(createProductsListForMock());
+
+        //when
+        InvalidSellerException exception = assertThrows(InvalidSellerException.class,
+                () -> orderHelper.prepareOrderToSave(orderDTO, order));
+
+        //then
+        assertEquals(expectedMessage, exception.getMessage());
+    }
+
+    @Test
+    void shouldThrowExceptionBecauseInOrderThereIsProductIdWithIsNotInDB() {
+        //given
+        Long incorrectId = 4L; //more than 3
+        OrderRequestDTO orderDTO = createOrderRequestDTOWithIncorrectProductId(incorrectId);
+        Order order = createOrderInStageAfterMapping();
+        String expectedMessage = "We can't find products with id = " + incorrectId;
+
+        when(productRepository.findAllById(Mockito.any())).thenReturn(createProductsListForMock());
+
+        //when
+        ProductsNotFoundException exception = assertThrows(ProductsNotFoundException.class,
+                () -> orderHelper.prepareOrderToSave(orderDTO, order));
+
+        //then
+        assertEquals(expectedMessage, exception.getMessage());
+    }
+
+    @Test
+    void shouldThrowExceptionBecauseInOrderThereIsWrongQuantity() {
+        //given
+        OrderLineDTO orderLineWithWrongQuantity = new OrderLineDTO(3L, 100);
+        OrderRequestDTO orderDTO = createOrderRequestDTOWithWrongQuantity(orderLineWithWrongQuantity);
+        Order order = createOrderInStageAfterMapping();
+        String expectedMessage = "There is incorrect data about stock of this items = 3";
+
+        when(productRepository.findAllById(Mockito.any())).thenReturn(createProductsListForMock());
+
+        //when
+        InsufficientStockException exception = assertThrows(InsufficientStockException.class,
+                () -> orderHelper.prepareOrderToSave(orderDTO, order));
+
+        //then
+        assertEquals(expectedMessage, exception.getMessage());
+    }
+
+
+    private void setUpSecurityContextForTests(boolean isAuthenticated) {
+        SecurityContext securityContext = Mockito.mock(SecurityContext.class);
+        Authentication authentication = Mockito.mock(Authentication.class);
+
+        SecurityContextHolder.setContext(securityContext);
```

Comment:
To jest kiepskie, ale jak będziesz miał beana z metodką od zwracania username zautoryzowanego usera to nie powinno być potrzeby robić takich brzydkich rzeczy:D